src/index.rs

```diff
@@ -188,6 +188,40 @@ impl<T> BitcoinCoreRpcResultExt<T> for Result<T, bitcoincore_rpc::Error> {
   }
 }
 
+/// An event that indicates that the location of an inscription has changed.
+/// Newly created inscriptions will include additional metadata including
+/// rarity, cursed status, charms, etc.
+#[derive(Debug, Clone)]
+pub enum LocationUpdateEvent {
```

Comment:
Let's make the name just `Event`, in case we want to add events that aren't related to inscriptions.

```suggestion
#[derive(Debug, Clone)]
pub enum Event {
```

Reply:
Let's also move this into a submodule, `src/index/event.rs`